.github/workflows/ci.yaml

```
@@ -6,6 +6,10 @@ on:
   pull_request:
     branches: [main]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

```suggestion
  group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
```
See comments from `BjornO` in the stackoverflow thread you linked. Including `github.run_id` as a fallback may be overkill here since we don't have a `workflow_dispatch` trigger, but I figure why not.